unyt/_array_functions.py

```diff
@@ -13,6 +13,37 @@
     UnytError,
 )
 
+# Functions that are wrappable but don't really make sense with units
+_UNSUPPORTED_FUNCTIONS = {
+    # Polynomials
+    np.poly,
+    np.polyadd,
+    np.polyder,
+    np.polydiv,
+    np.polyfit,
+    np.polyint,
+    np.polymul,
+    np.polysub,
+    np.polyval,
+    np.roots,
+    np.vander,
+    # IO functions (no way to add units)
+    np.save,
+    np.savez,
+    np.savez_compressed,
+    # datetime64 is not a sensible dtype for unyt_array
+    np.datetime_as_string,
+    np.busday_count,
+    np.busday_offset,
+    np.is_busday,
+    # not clear how to approach
+    np.piecewise,  # astropy.units doens't have a simple implementation either
+    np.packbits,
+    np.unpackbits,
+    np.i0,
+    np.ix_,
```

Comment:
Can we split this into functions that are truly unsupported and we want to raise errors and functions where we're fine with numpy ignoring the units?

`np.save` ignoring units might be fine for some existing use cases, especially if someone were following our docs which explicitly gave `np.save` as an example of something you can do with a `unyt_array`. I don't really see a reason to break that if it used to work, even if it might be slightly less confusing for future users. If we really want to break it I'd prefer if we did a deprecation cycle first.

In addition to `np.save`, `np.savez` and `np.savez_compressed`, I don't think it really hurts anything if `np.i0` ignored units (e.g. like `np.sin` does and will continue to do).

I think it's fine for all the other ones to error and it's unlikely that someone has used them with a `unyt_array` already, but if you want to go over the list and feel like it might make sense for some of them to work without units, feel free to depart from what I'm suggesting.

Reply:
Sure, I can make this more subtle. Will edit in a minute.

Reply:
done. I've also left `np._ix` out of the PR to avoid conflicts with #398